src/compiler/types.ts

```diff
@@ -1807,23 +1850,37 @@ namespace ts {
         /* @internal */
         ContainsObjectLiteral   = 0x00400000,  // Type is or contains object literal type
         /* @internal */
-        ContainsAnyFunctionType = 0x00800000,  // Type is or contains object literal type
-        ESSymbol                = 0x01000000,  // Type of symbol primitive introduced in ES6
-        ThisType                = 0x02000000,  // This type
+        // ContainsAnyFunctionType = 0x00800000,  // Type is or contains object literal type
+        ESSymbol                = 0x00800000,  // Type of symbol primitive introduced in ES6
+        ThisType                = 0x01000000,  // This type
+
+        I8                      = 0x02000000,
+        U8                      = 0x04000000,
+        I16                     = 0x08000000,
+        U16                     = 0x10000000,
+        I32                     = 0x20000000,
+        U32                     = 0x40000000,
+        Float                   = 0x40000000,
 
         /* @internal */
-        Intrinsic = Any | String | Number | Boolean | ESSymbol | Void | Undefined | Null,
+        Intrinsic = Any | String | Number | Boolean | I8 | U8 | I16 | U16 | I32 | U32 | Float | ESSymbol | Void | Undefined | Null,
         /* @internal */
-        Primitive = String | Number | Boolean | ESSymbol | Void | Undefined | Null | StringLiteral | Enum,
+        Primitive = String | Number | I8 | U8 | I16 | U16 | I32 | U32 | Float | Boolean | ESSymbol | Void | Undefined | Null | StringLiteral | Enum,
         StringLike = String | StringLiteral,
-        NumberLike = Number | Enum,
-        ObjectType = Class | Interface | Reference | Tuple | Anonymous,
+        NumberLike = Number | Enum | I8 | U8 | I16 | U16 | I32 | U32 | Float,
+        ObjectType = Struct | Class | Interface | Reference | Tuple | Anonymous,
+
+        PrimitiveType = I8 | U8 | I16 | U16 | I32 | U32 | Float,
+        DoublePrecisionFloat = Number,
+        SinglePrecisionFloat = Float,
+        RealNumber = Number | Float,
+
         UnionOrIntersection = Union | Intersection,
         StructuredType = ObjectType | Union | Intersection,
         /* @internal */
         RequiresWidening = ContainsUndefinedOrNull | ContainsObjectLiteral,
         /* @internal */
-        PropagatingFlags = ContainsUndefinedOrNull | ContainsObjectLiteral | ContainsAnyFunctionType
+        PropagatingFlags = ContainsUndefinedOrNull | ContainsObjectLiteral,

```

Comment:
Is the removal of `ContainsAnyFunctionType` from `PropagatingFlags` intentional?